code/datums/helper_datums/butchering.dm

```diff
@@ -349,3 +352,249 @@ var/global/list/animal_butchering_products = list(
 #undef TEETH_BUNCH
 #undef TEETH_LOTS
 #undef TEETH_HUMAN
+
+/////////////////////////////////Butcher procs/////////////////////////////////////////////////
+
+//Unlike the previous butchering version the butcher_product datums no longer need to be in a global list to work.
+//Instead you place them directly in the mob's butcher_products list.
+//butcherCarveStep() can be used to change how long meat takes to butcher from individual mobs
+//butcherMeat() and butcherProduct() are able to be called without the rest of the process.
+//Call butcherProduct() with mod = TRUE to have it search for children of the butcher_product you passed to it in the mob's product list.
+//modMeat() and modProduct() can be easily used to add properties to butcher results based off the mob it came from.
+
+#define BUTCHER_MEAT 1
+#define BUTCHER_SPEED "bSpeed"
+#define BUTCHER_TOOL "bTool"
+#define BUTCHER_TNAME "bTName"
+
+/mob/living/proc/butcher()
+	set category = "Object"
+	set name = "Butcher"
+	set src in oview(1)
+
+	var/mob/living/user = usr
+	if(!butcherCheck(user))
+		return
+	var/butcherTool = null
+	var/toolName = null
+	var/speed_mod = 0
+	var/list/butchValues = butcherValueStep(user)
+	if(!butchValues?.len)
+		return
+	if(butchValues[BUTCHER_SPEED])
+		speed_mod = butchValues[BUTCHER_SPEED]
+	if(butchValues[BUTCHER_TOOL])
+		butcherTool = butchValues[BUTCHER_TOOL]
+	if(butchValues[BUTCHER_TNAME])
+		toolName = butchValues[BUTCHER_TNAME]
+	var/list/butcherOptions = butcherMenuStep(user)
+	if(!butcherOptions?.len)
```

Comment:
```suggestion
	if(!length(butcherOptions))
```
length returns 0 if the list is null or has length 0

Reply:
Yeah but what's wrong with this

Reply:
it's just weird?

Reply:
Not as weird as `length()` IMO

Reply:
ok don't do this then